src/main/java/seedu/address/model/candidate/UniqueCandidateList.java

```diff
@@ -66,6 +68,14 @@ public void setCandidate(Candidate target, Candidate editedCandidate) {
             throw new DuplicateCandidateException();
         }
 
+        ObservableList<Candidate> internalListCopy = FXCollections.observableArrayList(internalList);
+
+        internalListCopy.set(index, editedCandidate);
+
+        if (!candidatesAreUnique(internalListCopy)) {
+            throw new CommandException(MESSAGE_DUPLICATE_CANDIDATE);
```

Comment:
sorry @domlimm just have a very quick question on hindsight and reviewing this piece of code --> would it still function if we swapped this to `throw new DuplicateCandidateException()`? might be a more appropriate exception to throw here! sorry as only noticed this upon reviewing it here..

Reply:
Ah yes for that. So if we look at `DuplicateCandidateException`, it throws a `RuntimeException`. I had the same thought but from the looks of it, I believe it's more suited towards the start up of the application (sync) instead of when we run commands (async). Thoughts? Good point @charmainehly!

Reply:
Alright thanks @domlimm for checking this out :)) appreciate it!!!

Reply:
You're most welcome! No biggie